chronos.py

```diff
@@ -0,0 +1,236 @@
+            ],
+            'MANUSCRIPT_FILES': [
+                cls.serialize_file(preprint, file_node)
+                for file_node in OsfStorageFile.objects.filter(node=preprint.node)
+            ],
+            'ABSTRACT': preprint.node.description,
+            'ARTICLE_TYPE': 'research-article',  # ??
+            'DOI': preprint.preprint_doi,
+            'MANUSCRIPT_TITLE': preprint.node.title,
+            'PROVIDER_MANUSCRIPT_ID': preprint._id,
+            'CHRONOS_JOURNAL_ID': journal_id,
+            'MANUSCRIPT_URL': preprint.url,
+
+            # 'CATEGORIES': '',
+            # 'CHRONOS_SUBMISSION_URL': '',
+            # 'CLASSIFICATIONS': '',
+            # 'INVOICES': '',
+            # 'JOURNAL_EISSN': '',
+            # 'JOURNAL_ISSN': '',
+            # 'JOURNAL_ISSUE': '',
+            # 'JOURNAL_NAME': '',
+            # 'JOURNAL_VOLUME': '',
+            # 'KEYWORDS': '',
+            # 'LICENSE': '',
+            # 'PMC_ID': '',
+            # 'PUBLISHER_JOURNAL_ID': '',
+            # 'PUBLISHER_NAME': '',
+            # 'RUNNING_HEAD': '',
+            'STATUS_CODE': status,
+        }
+
+    @classmethod
+    def serialize_user(cls, user):
+        return {
+            'CHRONOS_USER_ID': user.chronos_user_id,
+            'EMAIL': user.username,
+            'GIVEN_NAME': user.given_name,
+            'ORCID_ID': user.social.get('orcid', None),
+            'PARTNER_USER_ID': user._id,
+            'SURNAME': user.family_name,
+        }
+
+    @classmethod
+    def serialize_author(cls, contributor):
+        ret = cls.serialize_user(contributor.user)
+        if contributor._order == 0:
+            contribution = 'firstAuthor'
+        else:
+            contribution = 'submittingAuthor'
+        ret.update({
+            'CONTRIBUTION': contribution,
+            'ORGANIZATION': '',
+        })
+
+        return ret
+
+    @classmethod
+    def serialize_file(cls, preprint, file_node):
+        assert file_node.is_file
+
+        # if preprint.primary_file == file_node:
+        #     category = 'articleContent'
+        # else:
+        #     category = 'supplementaryMaterial'
+
+        return {
+            # 'FILE_DOWNLOAD_URL': file_node.deep_url,
+            'FILE_DOWNLOAD_URL': 'https://osf.io/2k3jp/download',
+            'FILE_NAME': file_node.name,
+            'MANUSCRIPT_FILE_CATEGORY': 'Publication Files',
+            # 'MANUSCRIPT_FILE_CATEGORY': category,
+        }
+
+class ChronosClient(object):
+
+    def __init__(self, username, password, api_key, host='http://sandbox.api.chronos-oa.com'):
+        self._client = ChronosRestClient(username, password, api_key, host=host)
+
+    def sync_journals(self):
+        journals = []
+        for journal in self._client.get_journals():
+            journals.append(ChronosJournal.objects.get_or_create(journal_id=journal['JOURNAL_ID'], defaults={
+                'raw_response': journal,
```

Comment:
Are we not parsing this / do we expect it never to change?

Reply:
`get_or_create` could make multiple instances of the same journal if the `raw_response` ever changes

Reply:
I lied, didn't realize it was wrapped with `defaults`. Carry on

Reply:
But also, should we update any fields that do change when we `sync_journals`?